src/Checks.js

```diff
@@ -90,18 +87,24 @@ class CheckForm extends Component {
     }
 
     fields.push(<SingleFieldElement 
-          label={label}
-          value={value}
-          controlId={controlId}
-          onChange={onChange}
-          placeholder={placeholder}
-          key='control'
-        />);
+                    label={label}
+                    value={value}
+                    controlId={controlId}
+                    onChange={onChange}
+                    placeholder={placeholder}
+                    key='control'
+                />);
 
     return (
       <ResourceForm data={this.state} baseResource={this.props.baseResource}>
+        <h2>New Check</h2>
+        <FormGroup controlID="checkName">
+          <ControlLabel>Name</ControlLabel>
+          <FormControl value={this.state.check_name} placeholder="Enter a name for your check" />
```

Comment:
In response to your first question, we always used Redux-Form which handled all of that for us.  Didn't really stop to see if you were using it.

As to your second, I didn't really know what that did and the rest seemed to have some kind of extra logic attached to them to do special shit (like show and hide).  I figured that's what those were for.